snapcraft/parts/lifecycle.py

```diff
@@ -35,54 +36,58 @@
     import argparse
 
 
-_PROJECT_FILES = [
-    Path("snapcraft.yaml"),
-    Path("snap/snapcraft.yaml"),
-    Path("build-aux/snap/snapcraft.yaml"),
-    Path(".snapcraft.yaml"),
+@dataclass
+class _SnapProject:
+    project_file: Path
+    project_dir: Path = Path("snap")
```

Comment:
This name could generate some confusion because `project_dir` is used elsewhere as the project top-level directory where you run snapcraft from. IIRC this directory (`snap` if you have `snap/snapcraft.yaml`) is called `assets_dir` in the rest of the code.

Reply:
fixed